### ISSUE TITLE
All instructions are skipping

### ISSUE THREAD

**Issue by atrauzzi** [debops/ansible-ifupdown#22]
I strongly suspect this is going to be a beginner issue, but I'm having a difficult time understanding why ansible isn't applying any actions I've configured from this role:

(I'm running ansible in pull-mode, so that's why I'm using localhost)

```

---
-
  hosts: localhost
  roles:
    - role: debops.ifupdown
      ifupdown_interfaces:
        -
          iface: wlan0
          inet: static
          auto: True
          options:
            address: 192.168.1.1
            netmask: 255.255.255.0
```

This results in the following output:

```
TASK: [debops.ifupdown | Check if default configuration is static] ************ 
skipping: [localhost]

TASK: [debops.ifupdown | Enable static default configuration if detected] ***** 
skipping: [localhost]

TASK: [debops.ifupdown | Divert original /etc/network/interfaces] ************* 
skipping: [localhost]

TASK: [debops.ifupdown | Create /etc/network/interfaces] ********************** 
skipping: [localhost]

TASK: [debops.ifupdown | Check if Network Manager is installed] *************** 
skipping: [localhost]

TASK: [debops.ifupdown | Enable Network Manager support if installed] ********* 
skipping: [localhost]

TASK: [debops.ifupdown | Load default interface configuration] **************** 
skipping: [localhost] => (item=/etc/ansible/roles/debops.ifupdown/vars/default.yml)

TASK: [debops.ifupdown | Prepare interface configuration if not defined] ****** 
skipping: [localhost]

TASK: [debops.ifupdown | Prepare interface configuration via dependency if defined] *** 
skipping: [localhost]

TASK: [debops.ifupdown | Add package dependencies for full interface declaration functionality] *** 
skipping: [localhost]

TASK: [debops.ifupdown | Create /etc/network/interfaces.d directory] ********** 
skipping: [localhost]

TASK: [debops.ifupdown | Delete interface configuration if requested] ********* 
skipping: [localhost] => (item={'auto': True, 'iface': 'wlan0', 'options': {'netmask': '255.255.255.0', 'address': '192.168.1.1'}, 'inet': 'static'})

TASK: [debops.ifupdown | Generate interface configuration] ******************** 
skipping: [localhost] => (item={'auto': True, 'iface': 'wlan0', 'options': {'netmask': '255.255.255.0', 'address': '192.168.1.1'}, 'inet': 'static'})

TASK: [debops.ifupdown | Reconfigure interfaces] ****************************** 
skipping: [localhost] => (item={u'skipped': True, u'changed': False})
```

As I mentioned, it's entirely likely I'm missing something, but would definitely appreciate any pointers or obvious-solutions! :D

**Comment by drybjed** [comment]
@atrauzzi I think that the format you used is wrong, try this instead:
```
---
- hosts: localhost
  sudo: True
  roles:
    - role: debops.ifupdown
      ifupdown_interfaces:
        - iface: wlan0
          inet: static
          auto: True
          options: |
            address: 192.168.1.1
            netmask: 255.255.255.0
```
Also, is `wlan0` present in `ansible_interfaces`?

**Comment by atrauzzi** [comment]
Would you happen to know where I check `ansible_interfaces`?

Also, I have the following and it's still skipping:

```
---
- 
  hosts: localhost
  roles:
    - role: debops.ifupdown
      ifupdown_interfaces:
        - iface: wlan0
          inet: static
          auto: True
          options:
            address: 192.168.1.1
            netmask: 255.255.255.0

  tasks:

    # Set up hostapd
    apt: name=hostapd
```

**Comment by drybjed** [comment]
You can check it by running

    ansible localhost -m setup -a 'filter=ansible_interfaces'

But I think I know what's up. One of the reasons `debops.ifupdown` would refrain from changing anything and skip all tasks would be that the system has enabled POSIX capabilities, and `cap_net_admin` is not granted to this user. Is this a OpenVZ/LXC container? To check this, switch to `root` account and run:

    capsh --print

You should either see an empty list of current capabilities (none are enabled, no restrictions), or a list of some capabilities and `cap_net_admin` missing.

**Comment by drybjed** [comment]
OH, one more think - you need `root.yml` playbook from the `debops/debops-playbooks` repository with some additional files and templates to prepare required Ansible facts on the host. That's probably what's missing here.

Or, add in inventory:

   ifupdown_ignore_cap12s: True

That should make it work.

**Comment by atrauzzi** [comment]
Haha, my situation is pretty unique, I'm actually using ansible to provision hardware.  Here's the output of the first command:

```
localhost | success >> {
    "ansible_facts": {
        "ansible_interfaces": [
            "lo", 
            "tunl0", 
            "wlan0", 
            "eth0"
        ]
    }, 
    "changed": false
}
```

**Comment by atrauzzi** [comment]
When you say "add in inventory", whereabouts is that?  Keep in mind, I don't have an inventory file, I'm just running `-i "localhost,"`

**Comment by drybjed** [comment]
@atrauzzi Not that unique I'm afraid... DebOps is made to provision hardware, look for example at `debops.hwraid` or `debops.ipxe` roles.

Well, then try this as a stop gap:

    ansible -i "localhost," --extra-args "ifupdown_ignore_cap12s=True" ...

And setup an inventory to distribute to that host, it will make configuration much easier since most of the variables in DebOps are lists or dicts.

**Comment by drybjed** [comment]
@atrauzzi Or, you can add `ifupdown_ignore_cap12s: True` with the other `ifupdown` options in the playbook, that should do the trick as well.

**Comment by atrauzzi** [comment]
Interesting, getting a new message on verbose:

**Comment by drybjed** [comment]
@atrauzzi Well, that's not much... Is that file present? Can you show the whole output of the role?

**Comment by atrauzzi** [comment]
Sorry, the rest seemed mostly uneventful, here it is:

```
TASK: [debops.ifupdown | Recalculate ifupdown based on POSIX capabilities] **** 
ok: [localhost] => {"ansible_facts": {"ifupdown": ""}}

TASK: [debops.ifupdown | Check if default configuration is static] ************ 
skipping: [localhost]

TASK: [debops.ifupdown | Enable static default configuration if detected] ***** 
skipping: [localhost]

TASK: [debops.ifupdown | Divert original /etc/network/interfaces] ************* 
skipping: [localhost]

TASK: [debops.ifupdown | Create /etc/network/interfaces] ********************** 
skipping: [localhost]

TASK: [debops.ifupdown | Check if Network Manager is installed] *************** 
skipping: [localhost]

TASK: [debops.ifupdown | Enable Network Manager support if installed] ********* 
skipping: [localhost]

TASK: [debops.ifupdown | Load default interface configuration] **************** 
skipping: [localhost] => (item=/etc/ansible/roles/debops.ifupdown/vars/default.yml)

TASK: [debops.ifupdown | Prepare interface configuration if not defined] ****** 
skipping: [localhost]

TASK: [debops.ifupdown | Prepare interface configuration via dependency if defined] *** 
skipping: [localhost]

TASK: [debops.ifupdown | Add package dependencies for full interface declaration functionality] *** 
skipping: [localhost]

TASK: [debops.ifupdown | Create /etc/network/interfaces.d directory] ********** 
skipping: [localhost]

TASK: [debops.ifupdown | Delete interface configuration if requested] ********* 
skipping: [localhost]

TASK: [debops.ifupdown | Generate interface configuration] ******************** 
skipping: [localhost]

TASK: [debops.ifupdown | Reconfigure interfaces] ****************************** 
skipping: [localhost] => (item=ifupdown_register_status.results)

TASK: [debops.ifupdown | Move interfaces out of the way before reversion] ***** 
ok: [localhost] => {"changed": false, "cmd": "rm -f /etc/network/interfaces", "rc": 0, "stderr": false, "stdout": "skipped, since /etc/network/interfaces.dpkg-divert does not exist"}

TASK: [debops.ifupdown | Revert original /etc/network/interfaces] ************* 
ok: [localhost] => {"changed": false, "cmd": "dpkg-divert --quiet --local --rename --remove /etc/network/interfaces", "rc": 0, "stderr": false, "stdout": "skipped, since /etc/network/interfaces.dpkg-divert does not exist"}
```

**Comment by drybjed** [comment]
```
TASK: [debops.ifupdown | Recalculate ifupdown based on POSIX capabilities] **** 
ok: [localhost] => {"ansible_facts": {"ifupdown": ""}}
```
This means that `debops.ifupdown` did not detect any POSIX capabilities at all (you don't have the fact script installed) so it refuses to perform any actions. Did you set `ifupdown_ignore_cap12s: True` in the variables?
```
TASK: [debops.ifupdown | Move interfaces out of the way before reversion] ***** 
ok: [localhost] => {"changed": false, "cmd": "rm -f /etc/network/interfaces", "rc": 0, "stderr": false, "stdout": "skipped, since /etc/network/interfaces.dpkg-divert does not exist"}

TASK: [debops.ifupdown | Revert original /etc/network/interfaces] ************* 
ok: [localhost] => {"changed": false, "cmd": "dpkg-divert --quiet --local --rename --remove /etc/network/interfaces", "rc": 0, "stderr": false, "stdout": "skipped, since /etc/network/interfaces.dpkg-divert does not exist"}
```
Due to the above issue (no cap12s detected), `debops.ifupdown` switched to a disabled state which means essentially reverting original /etc/network/interfaces. Since diverted file now doesn't exist, these tasks are OK.

You need to install [cap12s.fact](https://github.com/debops/debops-playbooks/blob/master/playbooks/files/etc/ansible/facts.d/cap12s.fact) local fact script on remote host, which is usually done by the [root.yml](https://github.com/debops/debops-playbooks/blob/master/playbooks/root.yml) playbook which is a required playbook in DebOps.

**Comment by atrauzzi** [comment]
Yeah, that last run was with the variable set.

**Comment by drybjed** [comment]
@atrauzzi Weird, I thought that would work... Can you try with the fact script? Also, what shows up when you run `capsh --print` ?

**Comment by atrauzzi** [comment]
Hm, that doesn't appear to be available on my platform (or I'm not having any luck finding the package it's installed from).  I'm running on an ARM debian derivative.

**Comment by drybjed** [comment]
Oh, I see. Hmm... That is peculiar then. How about a screen sharing session? Could you join our IRC channel (#debops @ FreeNode.net) so we can discuss it in real time?

**Comment by atrauzzi** [comment]
Sure thing.

**Comment by le9i0nx** [comment]
skip setting if there is a word static
I took into account a comment.
```
# This configuration file is auto-generated.
#
# WARNING: Do not edit this file, your changes will be lost.
# Please create/edit /etc/network/interfaces.head and
# /etc/network/interfaces.tail instead, their contents will be
# inserted at the beginning and at the end of this file, respectively.
#
# NOTE: it is NOT guaranteed that the contents of /etc/network/interfaces.tail
# will be at the very end of this file.
#

# Auto generated lo interface
auto lo
iface lo inet loopback

# Auto generated venet0 interface
auto venet0
iface venet0 inet manual
	up ifconfig venet0 up
	up ifconfig venet0 127.0.0.2
	up route add default dev venet0
	down route del default dev venet0
	down ifconfig venet0 down


iface venet0 inet6 manual
	up route -A inet6 add default dev venet0
	down route -A inet6 del default dev venet0

auto venet0:0
iface venet0:0 inet static
	address 8.8.8.8
	netmask 255.255.255.255
```

hostname en
```

PLAY [homes] ****************************************************************** 

GATHERING FACTS *************************************************************** 
ok: [74ru]
fatal: [lenovo] => SSH encountered an unknown error during the connection. We recommend you re-run the command using -vvvv, which will enable SSH debugging output to help diagnose the issue
ok: [en]

TASK: [debops.apt_preferences | Remove APT preferences] *********************** 
skipping: [74ru] => (item={'reason': 'Backport installed on Wheezy for version parity with Debian Jessie', 'by_role': 'debops.tinc', 'backports': ['wheezy'], 'package': 'tinc'})
skipping: [en] => (item={'reason': 'Backport installed on Wheezy for version parity with Debian Jessie', 'by_role': 'debops.tinc', 'backports': ['wheezy'], 'package': 'tinc'})

TASK: [debops.apt_preferences | Create APT preferences] *********************** 
skipping: [en] => (item={'reason': 'Backport installed on Wheezy for version parity with Debian Jessie', 'by_role': 'debops.tinc', 'backports': ['wheezy'], 'package': 'tinc'})
skipping: [74ru] => (item={'reason': 'Backport installed on Wheezy for version parity with Debian Jessie', 'by_role': 'debops.tinc', 'backports': ['wheezy'], 'package': 'tinc'})

TASK: [debops.ferm | Calculate debconf answer] ******************************** 
ok: [en]
ok: [74ru]

TASK: [debops.ferm | Configure ferm status in debconf] ************************ 
ok: [74ru]
ok: [en]

TASK: [debops.ferm | Ensure ferm is installed] ******************************** 
ok: [74ru] => (item=ferm)
ok: [en] => (item=ferm)

TASK: [debops.ferm | Create configuration directories] ************************ 
ok: [74ru] => (item=/etc/ferm/ferm.d)
ok: [74ru] => (item=/etc/ferm/filter-input.d)
ok: [en] => (item=/etc/ferm/ferm.d)
ok: [en] => (item=/etc/ferm/filter-input.d)

TASK: [debops.ferm | Configure ferm default variables] ************************ 
ok: [74ru]
ok: [en]

TASK: [debops.ferm | Configure main ferm config file] ************************* 
ok: [74ru]
ok: [en]

TASK: [debops.ferm | Allow SSH access from Ansible Controller] **************** 
ok: [74ru]
ok: [en]

TASK: [debops.ferm | Configure forwarding in ip(6)tables if enabled] ********** 
ok: [74ru]
ok: [en]

TASK: [debops.ferm | Remove ip(6)tables rules if requested] ******************* 
skipping: [74ru]
skipping: [en]

TASK: [debops.ferm | Configure ip(6)tables rules] ***************************** 
skipping: [74ru]
skipping: [en]

TASK: [debops.ferm | Remove iptables INPUT rules if requested] **************** 
skipping: [74ru] => (item={'protocol': ['tcp', 'udp'], 'accept_any': u'True', 'weight': '50', 'dport': ['tinc'], 'type': 'dport_accept', 'filename': 'tinc_dependency_accept'})
skipping: [en] => (item={'protocol': ['tcp', 'udp'], 'accept_any': u'True', 'weight': '50', 'dport': ['tinc'], 'type': 'dport_accept', 'filename': 'tinc_dependency_accept'})

TASK: [debops.ferm | Configure iptables INPUT rules] ************************** 
ok: [74ru] => (item={'protocol': ['tcp', 'udp'], 'accept_any': u'True', 'weight': '50', 'dport': ['tinc'], 'type': 'dport_accept', 'filename': 'tinc_dependency_accept'})
ok: [en] => (item={'protocol': ['tcp', 'udp'], 'accept_any': u'True', 'weight': '50', 'dport': ['tinc'], 'type': 'dport_accept', 'filename': 'tinc_dependency_accept'})

TASK: [debops.ferm | Apply iptables rules if ferm is enabled] ***************** 
ok: [74ru]
ok: [en]

TASK: [debops.ferm | Clear iptables rules if ferm is disabled] **************** 
skipping: [en]
skipping: [74ru]

TASK: [debops.ferm | Configure sysctl] **************************************** 
ok: [74ru]
ok: [en]

TASK: [debops.ferm | Ensure that /etc/network/if-pre-up.d exists] ************* 
ok: [74ru]
ok: [en]

TASK: [debops.ferm | Configure forwarding in ifupdown if enabled] ************* 
ok: [74ru]
ok: [en]

TASK: [debops.ferm | Ensure that Ansible local facts directory exists] ******** 
ok: [74ru]
ok: [en]

TASK: [debops.ferm | Save ferm local facts] *********************************** 
ok: [74ru]
ok: [en]

TASK: [debops.ifupdown | Recalculate ifupdown based on POSIX capabilities] **** 
ok: [en]
ok: [74ru]

TASK: [debops.ifupdown | Check if default configuration is static] ************ 
ok: [74ru]
ok: [en]

TASK: [debops.ifupdown | Enable static default configuration if detected] ***** 
skipping: [74ru]
ok: [en]

TASK: [debops.ifupdown | Divert original /etc/network/interfaces] ************* 
skipping: [en]
ok: [74ru]

TASK: [debops.ifupdown | Create /etc/network/interfaces.d directory] ********** 
skipping: [en]
ok: [74ru]

TASK: [debops.ifupdown | Provide original interfaces temporarily] ************* 
skipping: [en]
skipping: [74ru]

TASK: [debops.ifupdown | Remove redundant configuration from original interfaces] *** 
skipping: [en] => (item=^source /etc/network/interfaces.d/*)
skipping: [en] => (item=^auto lo)
skipping: [en] => (item=^iface lo inet loopback)
skipping: [74ru] => (item=^source /etc/network/interfaces.d/*)
skipping: [74ru] => (item=^auto lo)
skipping: [74ru] => (item=^iface lo inet loopback)

TASK: [debops.ifupdown | Create /etc/network/interfaces] ********************** 
skipping: [en]
ok: [74ru]

TASK: [debops.ifupdown | Check if Network Manager is installed] *************** 
ok: [74ru]
ok: [en]

TASK: [debops.ifupdown | Enable Network Manager support if installed] ********* 
skipping: [en]
skipping: [74ru]

TASK: [debops.ifupdown | Load default interface configuration] **************** 
skipping: [en] => (item=/home/le9i0nx/crypt/ansible/roles/debops.ifupdown/vars/default_static.yml)
ok: [74ru] => (item=/home/le9i0nx/crypt/ansible/roles/debops.ifupdown/vars/default.yml)

TASK: [debops.ifupdown | Prepare interface configuration if not defined] ****** 
skipping: [en]
skipping: [74ru]

TASK: [debops.ifupdown | Prepare interface configuration via dependency if defined] *** 
skipping: [en]
ok: [74ru]

TASK: [debops.ifupdown | Gather required network interface capabilities] ****** 
skipping: [en]
ok: [74ru]

TASK: [debops.ifupdown | Add package dependencies for full interface declaration functionality] *** 
skipping: [en]
ok: [74ru] => (item=resolvconf)

TASK: [debops.ifupdown | Create /etc/network/interfaces.d directory] ********** 
skipping: [en]
ok: [74ru]

TASK: [debops.ifupdown | Create /etc/network/interfaces.config.d directory] *** 
skipping: [en]
ok: [74ru]

TASK: [debops.ifupdown | Install network reconfiguration script] ************** 
skipping: [en]
ok: [74ru]

TASK: [debops.ifupdown | Delete interface configuration if requested] ********* 
skipping: [en]
skipping: [74ru] => (item={'iface': u'mesh0', 'force': True, 'addresses': ['172.16.35.10/24'], 'weight': '75', 'auto': u'True', 'type': 'interface', 'options': u'tinc-net mesh0\ntinc-chroot yes\ntinc-mlock yes\ntinc-user tinc-vpn', 'inet': u'static'})

TASK: [debops.ifupdown | Generate interface configuration] ******************** 
skipping: [en]
ok: [74ru] => (item={'iface': u'mesh0', 'force': True, 'addresses': ['172.16.35.10/24'], 'weight': '75', 'auto': u'True', 'type': 'interface', 'options': u'tinc-net mesh0\ntinc-chroot yes\ntinc-mlock yes\ntinc-user tinc-vpn', 'inet': u'static'})

TASK: [debops.ifupdown | Create random temporary directory for ifupdown data] *** 
skipping: [en]
skipping: [74ru]

TASK: [debops.ifupdown | Generate list of changed interfaces in temporary file] *** 
skipping: [en]
skipping: [74ru]

TASK: [debops.ifupdown | Generate config script in a known location] ********** 
skipping: [en]
skipping: [74ru]

TASK: [debops.ifupdown | Reconfigure network interfaces] ********************** 
skipping: [en]
skipping: [74ru]

TASK: [debops.ifupdown | Move interfaces out of the way before reversion] ***** 
skipping: [en]
skipping: [74ru]

TASK: [debops.ifupdown | Revert original /etc/network/interfaces] ************* 
skipping: [74ru]
skipping: [en]

TASK: [debops.tinc | Install required packages] ******************************* 
ok: [74ru]
ok: [en]

TASK: [debops.tinc | Create system group for VPN service] ********************* 
ok: [74ru]
ok: [en]

TASK: [debops.tinc | Create system user for VPN service] ********************** 
ok: [74ru]
ok: [en]

TASK: [debops.tinc | Create required directories] ***************************** 
ok: [74ru]
ok: [en]

TASK: [debops.tinc | Set tincd ulimit options] ******************************** 
ok: [74ru]
ok: [en]

TASK: [debops.tinc | Generate main configuration file] ************************ 
ok: [74ru]
ok: [en]

TASK: [debops.tinc | Generate network scripts when defined] ******************* 
skipping: [74ru]
skipping: [en]

TASK: [debops.tinc | Initialize RSA key pairs] ******************************** 
ok: [74ru]
ok: [en]

TASK: [debops.tinc | Create persistent copy of host public key] *************** 
ok: [74ru]
ok: [en]

TASK: [debops.tinc | Generate host configuration file] ************************ 
ok: [74ru]
ok: [en]

TASK: [debops.tinc | Assemble host configuration file from parts] ************* 
ok: [74ru]
ok: [en]

TASK: [debops.tinc | Read host configuration file into a variable] ************ 
ok: [74ru]
ok: [en]

TASK: [debops.tinc | Redistribute configuration of internal hosts] ************ 
ok: [en -> 74ru] => (item=74ru)
skipping: [en -> 74ru] => (item=en)
skipping: [en -> 74ru] => (item=lenovo)
skipping: [74ru] => (item=74ru)
ok: [74ru -> en] => (item=en)
skipping: [74ru -> en] => (item=lenovo)

TASK: [debops.tinc | Add configuration of external hosts] ********************* 
skipping: [74ru]
skipping: [en]

TASK: [debops.tinc | Start VPN interface if required] ************************* 
ok: [74ru]
ok: [en]

PLAY RECAP ******************************************************************** 
debops.tinc | Install required packages --------------------------------- 5.40s
debops.ferm | Create configuration directories -------------------------- 5.24s
debops.ferm | Apply iptables rules if ferm is enabled ------------------- 3.63s
debops.ferm | Ensure ferm is installed ---------------------------------- 3.41s
debops.ferm | Configure forwarding in ifupdown if enabled --------------- 3.29s
debops.ferm | Save ferm local facts ------------------------------------- 3.28s
debops.tinc | Generate main configuration file -------------------------- 3.27s
debops.tinc | Set tincd ulimit options ---------------------------------- 3.27s
debops.tinc | Generate host configuration file -------------------------- 3.25s
debops.ferm | Ensure that Ansible local facts directory exists ---------- 3.15s
           to retry, use: --limit @/home/le9i0nx/test.tinc.retry

74ru                       : ok=48   changed=0    unreachable=0    failed=0   
en                         : ok=38   changed=0    unreachable=0    failed=0   
lenovo                     : ok=0    changed=0    unreachable=1    failed=0   

```

**Comment by drybjed** [comment]
@le9i0nx Yeah, IIRC the issue was the `static` configuration in `/etc/network/interfaces`, we resolved that over IRC.

**Comment by le9i0nx** [comment]
Is it possible to force to generate files in the folder(/etc/network/interfaces.d/*), regardless of file(/etc/network/interfaces) contents?
I can make a patch, but it will take some time.

**Comment by drybjed** [comment]
@le9i0nx [It is now](https://github.com/debops/ansible-ifupdown/blob/master/defaults/main.yml#L35-L40) :-) Let me know if it works as expected, I don't have static hosts handy to test.

**Comment by le9i0nx** [comment]
It does not create a file in the folder /etc/network/interfaces.d/
ansible conclusion does not differ from the control host with dhcp

**Comment by drybjed** [comment]
@le9i0nx I see... And you set the variable to `True`? Then I guess I'll need to test this, unless you find out what's happening. Thanks for the heads up.

**Comment by le9i0nx** [comment]
ifupdown_ignore_static: True

**Comment by drybjed** [comment]
@le9i0nx Yes, that should do it.

**Comment by le9i0nx** [comment]
I do not see in your code when it should be copying from /etc/network/interfaces.config.d/ in /etc/network/interfaces.d/
This is not at all apparent.
which line carries Dunn action?

**Comment by drybjed** [comment]
This is done by a [separate script](https://github.com/debops/ansible-ifupdown/blob/master/templates/usr/local/lib/ifupdown-reconfigure-interfaces.j2#L136-L137) which is executed on remote host at the end of `debops.ifupdown`. It is done that way so that interfaces can be correctly brought down with current configuration and then back up with new. Ansible cannot really do that when networking is down, you know... :-)

**Comment by ypid** [comment]
I suspect that this issue could be fixed with the new v0.3.0 release of the role and could be closed.

**Comment by drybjed** [comment]
Yes, role does not have autodetection of the environment anymore, all tasks skipping shouldn't happen now.
